src/network/connection.cpp

```diff
@@ -62,18 +62,31 @@ namespace con
 
 #define PING_TIMEOUT 5.0
 
-BufferedPacket makePacket(Address &address, const SharedBuffer<u8> &data,
+u16 BufferedPacket::getSeqnum() const
+{
+	if (size() < BASE_HEADER_SIZE + 3) {
+		//std::cout << (int)data[0xFFFFFFFFF] << std::endl;
+		throw ConnectionException("Packet too short to contain a seqnum");
+	}
```

Comment:
I'm pretty sure size is checked before, so this check could be removed and the method made `inline`

Reply:
Due to `readU16` this would require the inclusion of "serialize.h", which is a pretty large header after macro/preprocessor. Hence I decided to not inline for this function right away.

Reply:
That's okay, the compiler might do it anyway.